src/core/cgroup.c

```diff
@@ -1629,7 +1629,10 @@ const char *unit_get_realized_cgroup_path(Unit *u, CGroupMask mask) {
 }
 
 static const char *migrate_callback(CGroupMask mask, void *userdata) {
-        return unit_get_realized_cgroup_path(userdata, mask);
+        /* If not realized at all, migrate to root .
+         * It may happen if we're upgrading from older version that didn't clean up.
+         */
+        return unit_get_realized_cgroup_path(userdata, mask) ?: "";
```

Comment:
Hmm, I'm not sure how I feel about this one, since it's hard to specify that as an invariant which we can actually check at runtime. I'm not sure how risky it is, I need to look more at how we end up here.

Reply:
Is this a practical situation you actually ran into, or just a theoretical concern?

Reply:
The blkio root cgroup wasn't realized when I tested with v245 on TW. After I reexeced my modified version I `to_callback` returned NULL and that wasn't well accepted (assert crash).